applications/conceptual-model-editor/src/app/diagram/dialog/select-classes-dialog.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useState } from "react";
+import { useModelGraphContext } from "../context/model-context";
+import { isSemanticModelClass } from "@dataspecer/core-v2/semantic-model/concepts";
+import { useBaseDialog } from "../components/base-dialog";
+import { useConfigurationContext } from "../context/configuration-context";
+import { CreateButton } from "../components/dialog/buttons/create-button";
+import { CancelButton } from "../components/dialog/buttons/cancel-button";
+import { t } from "../application/";
+import { isSemanticModelClassUsage } from "@dataspecer/core-v2/semantic-model/usage/concepts";
+import { type EntityDetailSupportedType, EntityProxy } from "../util/detail-utils";
+import { tailwindColorToHex } from "~/app/utils/color-utils";
+
+const getDefaultColor = () => {
+    return "#069420";
+};
+
+export const useSelectClassesDialog = () => {
+    const { isOpen, open, close, BaseDialog } = useBaseDialog();
+    const [onSelectConfirmCallback, setOnSelectConfirmCallback] = useState<((newEntityID: string[]) => void) | undefined>(undefined);
+
+    const localOpen = (onSelectConfirmCallback?: (newEntityID: string[]) => void) => {
+        setOnSelectConfirmCallback(onSelectConfirmCallback);
+        open();
+    };
+
+    // TODO: Maybe could be optimized later, when I get to know React better
```

Comment:
Please do not put TODOs into the code, create an issue instead. As using both TODOs and Issues give us two places where we keep track of things to do. When not sure,  you can always ask other to help you with a decision.